Changed from classification to category

LeeSingChanYau opened the pull request:
Changed “Classification” to “Category” in task edit view.
How to test:
Admin Login → Other Links → Projects → Click WBS Icon by project chosen → Choose WBS → Add or Edit Task
Make sure that the field name is category and not classification

one-community commented:
Excellent, thank you Alan. Reminder to change the status to "Done" and add the PR # and link to the Bugs doc whenever you post your PRs

PrabhjotSembhi commented on src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx:
```diff
@@ -353,17 +352,15 @@ const AddTaskModal = props => {
-        }
-    }
-    else {
+      if (classificationMother) {
```
I suggest renaming the variable too. 
Please rename all classifications and its related variables to "categories"

PrabhjotSembhi commented on src/components/Projects/WBS/WBSDetail/AddTask/AddTaskModal.jsx:
```diff
@@ -606,11 +603,15 @@ const AddTaskModal = props => {
-                  <select value={classification} onChange={e => setClassification(e.target.value)} >
-                    {classificationOptions.map(cla =>{
-                      return <option value={cla.value} key={cla.value}>{cla.label}</option>
+                  <select value={classification} onChange={e => setClassification(e.target.value)}>
```
Everything seems fine. just rename variables.

PrabhjotSembhi reviewed (commented):
I have checked it. working fine. 

Please rename variables also. that would be great. 

You can see my review video here. 

https://user-images.githubusercontent.com/82662768/215276899-252ce472-50a0-4eeb-8dcb-71b1727bc924.mp4

RafaelCastilhoCastro commented:
Hi @LeeSingChanYau,

I tested the PR and the fields are renamed to Category in both Add Task and Edit Task forms.
As observed by @PrabhjotSembhi I'd recommend changing the variables names for consistency's sake.
Thanks.

AnishPandita commented:
Hey @LeeSingChanYau I went over the Pr and everything seems to work as intended.  
The code structure seems reasonable and well oriented. As suggested by others renaming the variables to something more appropriate would be nice.

https://user-images.githubusercontent.com/111321945/215285386-293a2d6c-4118-4d13-811b-ce1a2eb94272.mov

Gukal7 commented:
Tested the changes, everything working perfect. Like others said, variable name changes would be great in terms of code readability. 
Although, I would suggest if the sub categories could be clearly identifiable, like the row line between 3.8 and 4 could have more width/boldness, else it's difficult to understand what lies under what category apart from seeing the # number. Or maybe changing the color scheme for the sub categories.
 
<img width="1440" alt="Screenshot 2023-01-28 at 3 15 26 PM" src="https://user-images.githubusercontent.com/30505274/215295580-4f723386-202c-4fd7-b917-8eafd5f32297.png">

Filipe-Oliv commented:
Hi @LeeSingChanYau

I tested your PR and its focus is fine.

As a suggestion for other PR, as @Gukal7 told you, I think the subs are not clearly defined. You could change the font color of the subs, for example, or something that makes it clear that the directory is a sub, like a branch symbol next to the folder name.

[week 03 - 02-07-23 (6).webm](https://user-images.githubusercontent.com/114119668/217363156-d2dccb50-cd2b-48bc-a0ed-9b96dba4c0d5.webm)

LeeSingChanYau commented:
Hey @Filipe-Oliv @Gukal7, I made some changes to the PR. As requested I changed all variable names from classification to category. And also changed the task name that is displayed as subcategory. If a task is a subcategory like 1.1, it would display something like "Energy/Save-Energy". I am using backslash to be clear what is the parent category. Please let me know if this looks good.
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/67836976/218268284-aa50e952-b3e5-4401-9640-dfb3c933f981.png">

PrabhjotSembhi reviewed (commented):
This is good now

one-community commented:
If good, someone please approve this and then I'll move it to the Management review stage.

Filipe-Oliv reviewed (dismissed):
All right. Looks good.

RafaelCastilhoCastro commented:
@one-community Jae, this PR is approved.

one-community commented:
Thank you. Moving to Management review

agujjar2206 commented:
Thanks for submitting this PR! I tested the changes locally by checking out the branch and running the application, and everything appears to be working correctly for both Add and Edit Task. I did not encounter any issues during testing.

<img width="1440" alt="Classification to Category Add" src="https://user-images.githubusercontent.com/94432002/219514622-d6d7e401-c757-4d6d-9f17-2d8b2535e2a0.png">
<img width="1440" alt="Classification to Category Edit" src="https://user-images.githubusercontent.com/94432002/219514630-9e28f78b-b295-4fc8-a07d-e36e4d1facde.png">

rauleffting commented on src/components/Projects/WBS/WBSDetail/Task/Task.jsx:
```diff
@@ -25,6 +25,7 @@ const Task = props => {
   const userPermissions = props.state.auth.user?.permissions?.frontPermissions;
   useEffect(() => {
     setIsCopied(false);
+    console.log('props.filteredTasks:', props.filteredTasks);
```
Please, delete this debug output.

LeeSingChanYau replied:
@rauleffting thanks for the feedback! Fixed it, please review again.

PrabhjotSembhi replied:
@rauleffting Hi, can you please check If your requested changes have been made, please approve this. 
Thanks.

LeeSingChanYau replied:
@one-community @rauleffting I deleted the console.log.

rauleffting reviewed (changes requested):
It is working perfectly. However, a debug output was left on the code. Please delete. After that, I'll approve. Thank you very much!

Lunaliar commented:
@LeeSingChanYau this works on my end! 
I also have some housekeeping if you want to take care of any of it while we're working on these files.

In addTaskModal.jsx
1. lines 114 & 116 there are debug functions
2. lines 636, 660 & 684 have a second className prop
3. line 730 has two onClicks on the same component

In AddTaskModal.jsx and EditTaskModal.jsx onKeypress is depreciated and could be swapped to onKeyDown

In ImportTask.jsx and Task.jsx the props for the components could be spread for less redundancy

suigunfjc reviewed (approved):
Hi @LeeSingChanYau
I pulled and tested your codes, everything looks pretty good to me and I will approve it. Good Job!

ayush0904 commented:
Excellent work!!! I have tested it and it's working perfect.
A small suggestion that could be implemented in future. We can have a constant file where we can save the dropdown values and labels, so that for updating the labels and values to multiple places we only have to change the constant file.

tingedudu8 reviewed (approved):
Hi @LeeSingChanYau, everything looks good to me. The name has been changed, as well as the variable names. The subtasks' names has been ajusted to make the file structure much clearer. Thanks!
<img width="933" alt="Screen Shot 2023-02-21 at 6 32 57 PM" src="https://user-images.githubusercontent.com/70128933/220490166-6c44ddb1-2f4d-4a27-ace2-54ca825fd123.png">
